library/chacha20.c

```diff
@@ -516,6 +516,11 @@ static const size_t test_lengths[2] =
     375U
 };
 
+/* Make sure no other definition is already present. */
+#ifdef ASSERT
+#undef ASSERT
+#endif
```

Comment:
According to the C standard, `#undef ASSERT` is enough: it simply has no effect if `ASSERT` isn't a defined macro. Are there compilers that warn about `#undef`ining a macro that isn't defined?

Reply:
Correct, `#undef` only is enough.